test/squid_test.rb

```diff
@@ -0,0 +1,785 @@
+#! /usr/bin/env rspec
+
+require_relative "./test_helper"
+
+Yast.import "Squid"
+
+describe "Yast::Squid" do
+  subject(:squid) { Yast::Squid }
+
+  describe "#Write" do
+    subject(:squid) { Yast::SquidClass.new }
+
+    before do
+      allow(Yast::Progress).to receive(:New)
+      allow(Yast::Progress).to receive(:NextStage)
+      allow(Yast::Report).to receive(:Error)
+
+      allow(Yast2::SystemService).to receive(:find).with("squid").and_return(service)
+
+      allow(Yast::Mode).to receive(:auto) { auto }
+      allow(Yast::Mode).to receive(:commandline) { commandline }
+
+      allow(squid).to receive(:Abort).and_return(false)
+      allow(squid).to receive(:writeAllSettings).and_return(settings_written)
+      allow(squid).to receive(:writeFirewallSettings).and_return(firewall_settings_written)
+
+      squid.main
```

Comment:
why it is needed only for write and not for all tests?

Reply:
I didn't write the tests for `Write`. I just moved them from the old location.